### PR TITLE
[WIPTEST] Collection conversion of Tenant & Project

### PR DESCRIPTION
**Pull request by vatsalparekh**
Purpose or Intent
=================
Refactored and converted Tenant to Collection + test cases updated
{{ pytest: -v -k tenant cfme/tests/infrastructure/test_infra_quota.py cfme/tests/configure/test_access_control.py }}

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 6879a78713c11214dd2c9001517dfb48b9711cb7:

`cfme/configure/access_control.py`:
- :bangbang: Line 1279:1: **E302** *expected 2 blank lines, found 1*

`cfme/tests/configure/test_access_control.py`:
- :bangbang: Line 798:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 957:1: **E302** *expected 2 blank lines, found 1*
- :bangbang: Line 961:1: **E302** *expected 2 blank lines, found 1*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/configure/access_control.py [hunk 1]
```diff
@@ -1256,20 +1261,72 @@ def is_displayed(self):
...
+    """Collection class for Tenant"""
+
+    def __init__(self, appliance):
+        self.appliance = appliance
+
+    def instantiate(self, name, description=None, parent=None):
+        return Tenant(collection=self, name=name, description=description, parent_tenant=parent)
 
+    def get_root_tenant(self):
+        return Tenant(collection=self, name='My Company', _default=True)
```
Comment (quarckster) [inline cfme/configure/access_control.py]: **REQUIRED**
You should use `instantiate` method

File: cfme/tests/configure/test_access_control.py [hunk 2]
```diff
@@ -789,8 +789,21 @@ def test_user_change_password(appliance, request):
 
 
 # Tenant/Project test cases
+
+@pytest.fixture(scope="module")
+def tenantcollection(appliance):
+    tc = TenantCollection(appliance=appliance)
+    yield tc
```
Comment (quarckster) [inline cfme/tests/configure/test_access_control.py]: **REQUIRED**
```python
return TenantCollection(appliance=appliance)
```
or
```python
return appliance.get(TenantCollection)
```

File: cfme/tests/configure/test_access_control.py [hunk 3]
```diff
@@ -789,8 +789,21 @@ def test_user_change_password(appliance, request):
...
+
+
+@pytest.fixture(scope="module")
+def projectcollection(appliance):
+    pc = ProjectCollection(appliance=appliance)
+    yield pc
```
Comment (quarckster) [inline cfme/tests/configure/test_access_control.py]: Here as well.

File: cfme/tests/infrastructure/test_infra_quota.py [hunk 4]
```diff
@@ -45,41 +45,43 @@ def set_group_cpu():
     group.remove_tag("Quota - Max CPUs *", "2")
 
 
+@pytest.fixture(scope="module")
+def roottenant(appliance):
+    tc = TenantCollection(appliance)
+    yield tc.get_root_tenant()
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_infra_quota.py]: **REQURIED**
`return`

File: cfme/configure/access_control.py [hunk 5]
```diff
@@ -1237,6 +1238,10 @@ def is_displayed(self):
 class DetailsTenantView(ConfigurationView):
     """ Details Tenant View """
     toolbar = View.nested(AccessControlToolbar)
+    name = Text('Name')
+    description = Text('Description')
+    parent = Text('Parent')
+    table = Table('//*[@id="fieldset"]/table')
```
Comment (quarckster) [inline cfme/configure/access_control.py]: **REQUIRED**
This should be replaced by [`BaseListEntity`](https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L2414).

**Comment by dajoRH** [comment]
Lint report for commit 1a21466a96e0652a6110bb7e6a0c431bdf37d6be:

`cfme/configure/access_control.py`:
- :bangbang: Line 1362:24: **E203** *whitespace before ':'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/configure/access_control.py [hunk 6]
```diff
@@ -1237,6 +1239,10 @@ def is_displayed(self):
 class DetailsTenantView(ConfigurationView):
     """ Details Tenant View """
     toolbar = View.nested(AccessControlToolbar)
+    name = Text('Name')
+    description = Text('Description')
+    parent = Text('Parent')
+    table = BaseListEntity()
```
Comment (quarckster) [inline cfme/configure/access_control.py]: **REQUIRED**
I didn't find any usage of this `table` attribute. What's the purpose of it? Also you shouldn't use `BaseListEntity` directly, look how it's implemented in [`host_views.py`](https://github.com/ManageIQ/integration_tests/blob/master/cfme/common/host_views.py#L73)

File: cfme/configure/access_control.py [hunk 7]
```diff
@@ -1213,6 +1214,7 @@ def is_displayed(self):
 class AllTenantView(ConfigurationView):
     """ All Tenants View """
     toolbar = View.nested(AccessControlToolbar)
+    table = Table('//*[@id="records_div"]/table')
```
Comment (quarckster) [inline cfme/configure/access_control.py]: **REQUIRED**
This should use [entities](https://github.com/ManageIQ/integration_tests/blob/master/widgetastic_manageiq/__init__.py#L2630)

File: cfme/configure/access_control.py (hunk 6 above)
Comment (vatsalparekh) [reply cfme/configure/access_control.py]: This table contains info about what's the quota set and what's the current quota in use or exceeded. 
Future test cases are planned around it.

**Comment by dajoRH** [comment]
Lint report for commit c240babcd65ff77e287ba76fea0843b92e994cae:

Lint OK! :cake: :punch: :cookie: .

* 0b33a5995695afb505a16e2dce69312ad9c34b31
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Pull request closed without merging**